### PR TITLE
UX: Multichain: Display total account balance in account menu

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**
For our multichain effort, we want to show total balances in both the home screen as well as the account menu and any other place we have `AccountListItem` components.  This PR creates a `useAccountTotalFiatBalance` hook that returns tokens and the total fiat balance.

## Fixes
https://github.com/MetaMask/MetaMask-planning/issues/1426

## **Manual testing steps**

1. Start the extension with `MULTICHAIN=1 yarn start`
2. Have an account with an ETH balance and token balances
3. On the home screen, ensure the large displayed balance is the sum of ETH and token fiat balances
4. Open the Accounts Menu
5. Ensure that each in the menu displays an fiat value that is the sum of the relevant account

With the current build, there should be no changes.

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-extension/assets/46655/87132195-98bc-4ad1-8414-b0999daf3037

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/asset-list/asset-list.js [hunk 1]
```diff
@@ -95,66 +83,12 @@ const AssetList = ({ onClickAsset }) => {
     getIstokenDetectionInactiveOnNonMainnetSupportedNetwork,
   );
 
-  const contractExchangeRates = useSelector(
```
Comment (darkwing) [inline ui/components/app/asset-list/asset-list.js:98]: All of this code was essentially moved to the new `ui/hooks/useAccountTotalFiatBalance.js` hook.

File: ui/hooks/useAccountTotalFiatBalance.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  getValueFromWeiHex,
+  sumDecimals,
+} from '../../shared/modules/conversion.utils';
+import { getConversionRate } from '../ducks/metamask/metamask';
+import { formatCurrency } from '../helpers/utils/confirm-tx.util';
+import { getTokenFiatAmount } from '../helpers/utils/token-util';
+import { isEqualCaseInsensitive } from '../../shared/modules/string-utils';
+import { useTokenTracker } from './useTokenTracker';
+
+export const useAccountTotalFiatBalance = (
```
Comment (darkwing) [inline ui/hooks/useAccountTotalFiatBalance.js:19]: `ui/components/app/asset-list/asset-list.test.js` passing proves that moving that logic to here works flawlessly!

**Review by NidhiKJha — COMMENTED** [review]
This looks awesome! Before, I approve, two questions:
1. Should we include a test for this hook?
2. When the selected primary currency in Fiat, shows two values in Fiat, would it confuse users?
Screenshot 2023-10-05 at 6 15 04 PM

**Review by brad-decker — DISMISSED** [review]

**Comment by darkwing** [comment]
> This looks awesome! Before, I approve, two questions:
> 
> 1. Should we include a test for this hook?
> 2. When the selected primary currency in Fiat, shows two values in Fiat, would it confuse users?
>    Screenshot 2023-10-05 at 6 15 04 PM

1.  Test added!
2.  I got to work with @brad-decker to figure out the primary token value preference display!

**Review by vthomas13 — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**